script.module.inputstreamhelper/lib/inputstreamhelper.py

```diff
@@ -0,0 +1,556 @@
+        required_diskspace = int(arm_device['filesize']) + int(arm_device['zipfilesize'])
+        dialog = xbmcgui.Dialog()
+        if dialog.yesno(self._language(30001), self._language(30002)) and dialog.yesno(self._language(30001), self._language(30006).format(self.sizeof_fmt(required_diskspace))) and self._widevine_eula():
+            if self._os != 'Linux':
+                dialog.ok(self._language(30004), self._language(30019).format(self._os))
+                return False
+            if required_diskspace >= self._diskspace():
+                dialog.ok(self._language(30004),
+                          self._language(30018).format(self.sizeof_fmt(required_diskspace)))
+                return False
+            if not self._cmd_exists('fdisk') and not self._cmd_exists('parted'):
+                dialog.ok(self._language(30004), self._language(30020).format('fdisk', 'parted'))
+                return False
+            if not self._cmd_exists('mount'):
+                dialog.ok(self._language(30004), self._language(30021).format('mount'))
+                return False
+            if not self._cmd_exists('losetup'):
+                dialog.ok(self._language(30004), self._language(30021).format('losetup'))
+                return False
+
+            self._url = arm_device['url']
+            downloaded = self._http_request(download=True, message=self._language(30022))
+            if downloaded:
+                dialog.ok(self._language(30023), self._language(30024))
+                busy_dialog = xbmcgui.DialogBusy()
+                busy_dialog.create()
+
+                bin_filename = self._url.split('/')[-1].replace('.zip', '')
+                bin_path = os.path.join(self._temp_path(), bin_filename)
+                if not self._unzip(self._temp_path(), bin_filename) or not self._set_loop_dev() or not self._losetup(bin_path) or not self._mnt_loop_dev():
+                    self._cleanup()
+                    busy_dialog.close()
+                    dialog.ok(self._language(30004), self._language(30005))
+                else:
+                    self._extract_widevine_cdm_from_img()
+                    self._install_cdm()
+                    self._cleanup()
+                    if self._has_widevine_cdm():
+                        dialog.ok(self._language(30001), self._language(30003))
+                        busy_dialog.close()
+                        return True
+                    else:
+                        busy_dialog.close()
+                        dialog.ok(self._language(30004), self._language(30005))
+
+        return False
+
+    def _widevine_eula(self):
+        """Display the Widevine EULA and prompt user to accept it."""
+        if os.path.exists(os.path.join(self._cdm_path(), config.WIDEVINE_LICENSE_FILE)):
+            license_file = os.path.join(self._cdm_path(), config.WIDEVINE_LICENSE_FILE)
+            with open(license_file, 'r') as f:
```

Comment:
please check that file size is within reasonable limits

Reply:
I think this should be fine. Downloading the license from the x86 files doesn't require more than a few megabytes (2 to be exact).